### PR TITLE
Field -> SettingField

### PR DESCRIPTION
**Pull request by GitOnUp**
- Add `SettingField` as a wrapper around `Field`, which sets the metadata we need for schema export, plus Field settings appropriately
- Unify on one set of defaults
- Unify on ServerSettings field names rather than broken up between UI/server
  - As a result, suggestOutputType is no longer used

### CODE REVIEW

**Comment by GitOnUp** [comment]
Previous schema: https://gist.github.com/GitOnUp/2a8bcf22c599c0f79e2299e2e576aac7
New schema: https://gist.github.com/GitOnUp/8f927aa96810767fbd377fd57661955d

Some fields have moved around, some defaults differ (added some that were missing).

File: src/schema/server_settings.py [hunk 1]
```diff
@@ -139,25 +139,73 @@ class Difficulty(str, Enum):
...
+        default: Optional[Any],
+        label: str,
+        description: str,
+        type: str,  # todo make enum
+        listof: Optional[str] = None,  # todo make enum
+        options: Optional[List[Dict]] = None,  # todo make inflated class
+        include_dynamic_options: Optional[str] = None,  # TODO enum with image-themes
+        required: Optional[bool] = None,
+        unused: Optional[bool] = None,
+        list_schema: Optional[List[Dict]] = None,  # how to represent this?
```
Comment (eob) [inline src/schema/server_settings.py:152]: Re: the list_schema comment -- Could it be another Pydantic class that's also using SettingField objects? That'd be pretty rad.

File: src/schema/server_settings.py [hunk 2]
```diff
@@ -139,25 +139,73 @@ class Difficulty(str, Enum):
...
+        variables_permitted: Optional[Dict[str, str]] = None,
+        min: Optional[Union[int | float]] = None,
+        **kwargs
+) -> Any:
+    """
+    Pass along parameters into the pydantic-native way of expressing them while also expressing them as our schema type.
+    `name` will be calculated from __fields__ later.
+    """
+    if "meta_setting" in kwargs:
+        raise ValueError("`meta_setting` is calculated, but was provided in arguments to SettingField.")
```
Comment (eob) [inline src/schema/server_settings.py:166]: Nice touch

File: src/schema/server_settings.py [hunk 3]
```diff
@@ -173,280 +221,182 @@ def setting_schema(cls, name: str) -> Dict[str, Any]:
...
-            "listof": "text",
-        },
+        label="Tags",
+        description="A list of short string tags.",
+        type="tag-list",
+        listof="text"
     )
     """For use on the profile marketing page and also during 'Magic Create' mode in the editor."""
 
+    # TODO what to do about this one?  It's not currently exported.
```
Comment (eob) [inline src/schema/server_settings.py:259]: We could def export it here -- IIRC it's in the schema file but probably just wasn't yet ported to the new Field style of describing it

**Review by eob — APPROVED** [review]
A few comments on some comments you made, but this looks really awesome.

File: src/schema/server_settings.py (hunk 1 above)
Comment (GitOnUp) [reply src/schema/server_settings.py:152]: Yeah I imagine we'd probably do the same kind of schema mechanism (maybe generalized to a mixin).  Will let that come in in a later PR.

File: src/schema/server_settings.py (hunk 3 above)
Comment (GitOnUp) [reply src/schema/server_settings.py:259]: hm, I'd have expected it to show up in the endpoint result diff I've been doing at some point if it had dropped out but I don't remember seeing that

**Pull request merged**